mlir/lib/Dialect/PDL/IR/Builtins.cpp

```diff
@@ -99,6 +99,25 @@ LogicalResult unaryOp(PatternRewriter &rewriter, PDLResultList &results,
             getIntegerAsAttr(APSInt(operandIntAttr.getValue(), false)));
       else
         results.push_back(getIntegerAsAttr(operandIntAttr.getAPSInt()));
+    } else if constexpr (T == UnaryOpKind::abs) {
+      if (integerType.isSigned()) {
+        // check overflow
+        if (operandIntAttr.getAPSInt() ==
+            APSInt::getMinValue(integerType.getIntOrFloatBitWidth(), false))
+          return failure();
+
+        results.push_back(rewriter.getIntegerAttr(
+            integerType, std::abs(operandIntAttr.getSInt())));
```

Comment:
isn't there a `abs` function that can be used? Hopefully it works with Signed and Signless integers.
```suggestion
            integerType, std::abs(operandIntAttr.getValue().abs())));
```